src/storyMap/components/StoryMapForm/MapLocationDialog.js

```diff
@@ -15,34 +15,55 @@
  * along with this program. If not, see https://www.gnu.org/licenses/.
  */
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
-import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
 import { Trans, useTranslation } from 'react-i18next';
 import {
-  Box,
   Button,
   Dialog,
   DialogActions,
   DialogContent,
   DialogTitle,
   Stack,
 } from '@mui/material';
-import mapboxgl from 'gis/mapbox';
-import { MAPBOX_ACCESS_TOKEN, MAPBOX_STYLE_DEFAULT } from 'config';
 import '@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css';
 import _ from 'lodash/fp';
-import {
-  MAPBOX_DEM_SOURCE,
-  MAPBOX_FOG,
-  MAPBOX_SKY_LAYER,
-} from 'storyMap/storyMapConstants';
+import MapboxGeocoder from 'gis/components/MapboxGeocoder';
+import MapboxMap, { useMap } from 'gis/components/MapboxMap';
+import MapboxMapControls from 'gis/components/MapboxMapControls';
 import { useStoryMapConfigContext } from './storyMapConfigContext';
 
+const MapLocationChange = props => {
+  const { onPositionChange } = props;
+  const { map } = useMap();
+
+  useEffect(() => {
+    if (!map) {
+      return;
+    }
+    const updatePosition = () => {
+      onPositionChange({
+        center: map.getCenter(),
+        zoom: map.getZoom(),
+        pitch: map.getPitch(),
+        bearing: map.getBearing(),
+      });
+    };
+    map.on('load', updatePosition);
+    map.on('move', updatePosition);
+
+    return () => {
+      map.off('load', updatePosition);
+      map.off('move', updatePosition);
+    };
+  }, [map, onPositionChange]);
+
+  return null;
```

Comment:
question: is this required? if so, should it match the return on line 40?

Reply:
Yes, a React component can return `undefined`